### PR TITLE
Extended sharing options

### PR DESCRIPTION
**Pull request by wujku**
Related to #70. 

Zrzut ekranu 2019-12-19 o 16 03 48

New options for saved search sharing: 
[+] Roles
[+] Share Globally

@fashxp Changes contains replaced Installer method, which extends MigrationInstaller instead of AbstractInstaller. 
I added one migration `Version20191218105114`, have you any idea how run this migration automatically after composer update?

### CODE REVIEW

**Comment by wujku** [comment]
@fashxp I simulated composer update with next version (v2.1.3) contains my changes, after update, package was disabled and has ability to install from Manage Extensions tab.

Zrzut ekranu 2019-12-19 o 16 19 28

Current users of package who have previously used it, after clicked install button, should see below log:

```
Installing bundle AdvancedObjectSearchBundle

Migrating up to 00000001 from 0

  ++ migrating 00000001

Migration 00000001 was executed but did not result in any SQL statements.

  ++ migrated (0.75s)

  ------------------------

  ++ finished in 0.75s
  ++ 1 migrations executed
  ++ 0 sql queries

######################################################################

Running AdvancedObjectSearchBundle updates after installation

Migrating up to 20191218105114 from 00000001

  ++ migrating 20191218105114

     -> ALTER TABLE bundle_advancedobjectsearch_savedsearch ADD shareGlobally TINYINT(1) DEFAULT NULL
     -> CREATE INDEX shareGlobally ON bundle_advancedobjectsearch_savedsearch (shareGlobally)

  ++ migrated (0.61s)

  ------------------------

  ++ finished in 0.61s
  ++ 1 migrations executed
  ++ 2 sql queries
```

Migration `00000001` specified as `AdvancedObjectSearchBundle\Installer::migrateInstall` has validation that checks if bundle tables has not been created before.

**Comment by fashxp** [comment]
add running migration automatically after composer update see this discussion https://github.com/pimcore/pimcore/issues/5445 ... any input is appreciated

**Comment by fashxp** [comment]
besides that, thank you very much! will have a look at it asap.

File: src/Resources/public/js/searchConfigPanel.js [hunk 1]
```diff
@@ -164,10 +178,40 @@ pimcore.bundle.advancedObjectSearch.searchConfigPanel = Class.create(pimcore.ele
...
+        var shareItems = [this.sharingField, this.rolesSharingField];
+
+        var user = pimcore.globalmanager.get("user");
+        if (user.admin) {
+            this.shareGlobally = new Ext.form.field.Checkbox(
+                {
+                    fieldLabel: t("share_globally"),
+                    inputValue: true,
+                    name: "share_globally",
+                    value: this.data.settings ? this.data.settings.shareGlobally : false
```
Comment (fashxp) [inline src/Resources/public/js/searchConfigPanel.js]: shared globally is not initialized properly when opening saved search, even it is set in DB row

**Comment by fashxp** [comment]
thx very much

**Pull request merged**